File: Tokens/dollartohousetoken/contracts/src/test/java/net/corda/samples/dollartohousetoken/contracts/StateTests.java
package net.corda.samples.dollartohousetoken.contracts;


import net.corda.samples.dollartohousetoken.states.HouseState;
import net.corda.testing.node.MockServices;
import org.junit.Test;
public class StateTests {
    private final MockServices ledgerServices = new MockServices();

    //sample State tests
    @Test
    public void hasConstructionAreaFieldOfCorrectType() throws NoSuchFieldException {
        // Does the message field exist?
        HouseState.class.getDeclaredField("constructionArea");
        // Is the message field of the correct type?
        assert(HouseState.class.getDeclaredField("constructionArea").getType().equals(String.class));
    }
}

File: kotlin-samples/Tokens/dollartohousetoken/contracts/src/test/kotlin/net/corda/samples/dollartohousetoken/contracts/StateTests.kt
package net.corda.samples.dollartohousetoken.contracts

import net.corda.samples.dollartohousetoken.states.HouseState
import net.corda.testing.node.MockServices
import org.junit.Test


class StateTests {
    private val ledgerServices = MockServices()

    //sample State tests
    @Test
    @Throws(NoSuchFieldException::class)
    fun hasConstructionAreaFieldOfCorrectType() {
        // Does the message field exist?
        HouseState::class.java.getDeclaredField("constructionArea")
        assert(HouseState::class.java.getDeclaredField("constructionArea").type == String::class.java)
    }
}